lib/run/src/Obelisk/Run.hs

```diff
@@ -191,26 +192,26 @@ runWidget toRun configs validFullEncoder = do
       -- specified in the route.
       port = fromMaybe 80 $ (_runApp_forceFrontendPort toRun)
                         <|> (fmap fromIntegral $ uri ^? uriAuthority . _Right . authPort . _Just)
-      -- This is the *actual* URI on which the frontend is served, i.e.
-      -- the URI from the route configuration but, possibly, with the
-      -- port we picked above. We need to compute this for two reasons:
-      --
-      --   1. The log. Self-explanatory.
-      --   2. JSaddle needs to know where the frontend is served.
-      actualUri = uri & uriAuthority . _Right . authPort ?~ fromInteger (fromIntegral port)
 
       -- This is the server that will handle the backend requests. We
       -- support shuttling them off to any host:port pair.
       redirectHost = _runApp_backendHost toRun
       redirectPort = _runApp_backendPort toRun
 
       beforeMainLoop = do
-        putStrLn $ "Frontend running on " ++ T.unpack (URI.render actualUri)
+        putStrLn $ "Frontend running on http://localhost:" ++ show port ++ "/"
+        when (routeIsTLS && isJust (_runApp_forceFrontendPort toRun)) $ do
+          putStrLn "Warning: Since a specific frontend port was requested, TLS will not be used for this session"
+          putStrLn "Please make sure that the public route is behind a reverse proxy to terminate TLS connections."
+
+      routeIsTLS = (Just "https" == uri ^? uriScheme . _Just . unRText)
+
       settings = setBeforeMainLoop beforeMainLoop (setPort port (setTimeout 3600 defaultSettings))
 
-      -- Providing TLS here will also incidentally provide it to proxied requests to the backend.
-      prepareRunner = case uri ^? uriScheme . _Just . unRText of
-        Just "https" -> do
+      -- Providing TLS here will also incidentally provide it to proxied
+      -- requests to the backend.
+      prepareRunner =
+        if routeIsTLS && isNothing (_runApp_forceFrontendPort toRun) then
```

Comment:
It's a little fraught that we are testing for TLS twice, here and in `beforeMainLoop`, once for logging and once for the actual logic. Is it easy to simplify that?